### PR TITLE
Enable Nette v3

### PR DESCRIPTION
**Pull request by trejjam**
BC Break? No

It relaxes dependencies to enable Nette v3 dependencies. Also, it handle differences in extension registration.

### CODE REVIEW

**Comment by trejjam** [comment]
Hi, can I do anything to make this or bit changed request to be merged?

**Comment by f3l1x** [comment]
Hi. Please make it compatible only for Nette 3. Thanks.

**Comment by trejjam** [comment]
Hi, I dropped support for pre Nette@v3. DI integration now uses strictly typed configuration object, except SentryLogger it should not be an issue. SentryLogger is not sealed, someone can extend it and expect an array as input instead of a configuration object. What about to make it final? - I am not using it I do not know what regular use case is.

**Comment by f3l1x** [comment]
Thanks. I need more time to explore it ;-)

**Comment by trejjam** [comment]
Sure, let me know. After l can update rest of the PR to the contributte in a same way.

**Comment by trejjam** [comment]
Hi, any feedback?

File: composer.json [hunk 1]
```diff
@@ -13,17 +13,17 @@
   ],
   "require": {
     "php": ">= 7.1",
-    "tracy/tracy": "~2.5.5"
+    "tracy/tracy": "~2.5.5|^2.6.2"
```
Comment (mabar) [inline composer.json]: ~

File: src/DI/Configuration/TracyConfiguration.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+use Nette\DI\Definitions\Statement;
+
+final class TracyConfiguration
+{
+
+	/** @var string */
+	public $logDir;
+
+	/** @var array[][]|string[]|Statement[]|null */
```
Comment (mabar) [inline src/DI/Configuration/TracyConfiguration.php]: `mixed[][]`

File: src/DI/TracyLoggingExtension.php [hunk 3]
```diff
@@ -71,20 +75,27 @@ public function beforeCompile(): void
...
-					$def = $builder->addDefinition($this->prefix('logger' . ($loggers++)));
-					Compiler::loadDefinition($def, $service);
+					$loggerName = 'logger' . ($loggers++);
+
+					$this->loadDefinitionsFromConfig(
+						[
+							$loggerName => $service,
+						]
+					);
+					$def = $builder->getDefinition($this->prefix($loggerName));
```
Comment (mabar) [inline src/DI/TracyLoggingExtension.php]: Can you please replace whole condition with contributte/di definitions helper?
https://contributte.org/blabs/2019/embedding-of-services-with-contributte-di.html

File: composer.json (hunk 1 above)
Comment (vody105) [reply composer.json]: ~2.5 za me

Comment (mabar) [reply composer.json]: We use `~x.y.z` format everywhere to ensure any minor version BC break in nette packages cannot break our own

Comment (vody105) [reply composer.json]: I'd use ~2.5 here because this package works with both 2.5 and 2.6.

Next tracy version is gonna be 3.0 imho

Comment (trejjam) [reply composer.json]: I am not sure about version 3.0, see: https://github.com/nette/tracy#installation

**Comment by f3l1x** [comment]
We've merged #28 sooner. Because it's minimal changes. We'll consider changes moving config to single value objects, but it needs time. 

Just for known.

**Pull request closed without merging**